presentation/src/main/java/me/androidbox/presentation/navigation/NavigationGraph.kt

```diff
@@ -83,7 +93,62 @@ fun NavigationGraph(
                 agendaScreenState = agendaScreenState,
                 agendaScreenEvent = { agendaScreenEvent ->
                     agendaViewModel.onAgendaScreenEvent(agendaScreenEvent)
-                })
+                },
+            onSelectedAgendaItem = {
+                /* TODO The item in the dropdown menu should be an enum or a sealed class that will determine which item was clicked */
+                navHostController.navigate(Screen.EventScreen.route)
+            })
+        }
+
+        /* Event Screen */
+        composable(
+            route = Screen.EventScreen.route
+        ) {
+            val eventViewModel: EventViewModel = hiltViewModel()
+            val eventScreenState by eventViewModel.eventScreenState.collectAsStateWithLifecycle()
+
+            EventScreen(
+                eventScreenState = eventScreenState,
+                eventScreenEvent = { eventScreenEvent ->
+                    eventViewModel.onEventScreenEvent(eventScreenEvent)
+                },
+                onEditTitleClicked = { title ->
+                    navHostController.navigate(route = "edit_screen/$title")
+                },
+                onEditDescriptionClicked = { description ->
+                    navHostController.navigate(Screen.EditScreen.route)
+                }
+            )
+        }
+        
+        /* Edit Screen */
+        composable(
+            route = Screen.EditScreen.route,
+            arguments = listOf(navArgument("content") {
+                type = NavType.StringType
+            })
+        ) {
+            val editScreenViewModel: EditScreenViewModel = hiltViewModel()
+            val editScreenState by editScreenViewModel.editScreenState.collectAsStateWithLifecycle()
+
+            /* TODO Just assigning an empty string for now if null just to get this working */
+            val content = it.arguments?.getString("content") ?: ""
+
+            LaunchedEffect(key1 = true) {
+                editScreenViewModel.onEditScreenEvent(
+                    editScreenEvent = EditScreenEvent.OnContentChanged(content))
+            }
```

Comment:
I think this is my main issue. Is the navigation part. I was trying to update the state and had to use the LaunchEffect as I think it was continuing to recompose. However, I think I am making a mess of this part

Reply:
What I did was:

When editing the title/description, I would save the new text on savedStateHandle

And when navigating back to, let's say, ReminderScreen
I would grab the key on the savedStateHandle, and pass it to the screen parameters
<img width="626" alt="image" src="https://user-images.githubusercontent.com/16141845/229208664-dc4a4bed-2d05-4fed-9fc1-8fcfe216cc95.png">

And inside the screen, if the title/description changed, I update the info using a LaunchedEffect 
<img width="388" alt="image" src="https://user-images.githubusercontent.com/16141845/229208765-75bb06e6-8f07-426c-b46d-e29c4b43b971.png">

Reply:
The key here is using `previousBackStackEntry`. Use this simplified sample together with the LaunchedEffect approach from Martin and it should work:

<img width="699" alt="Bildschirmfoto 2023-04-01 um 11 40 34" src="https://user-images.githubusercontent.com/53933333/229278548-093ce80d-41b6-43ee-8eda-ebd4187ce5a7.png">

Reply:
@philipplackner Thanks, previousBackStackEntry was the reason I spent many hours trying to get my solution to work.

Reply:
@philipplackner Thanks, previousBackStackEntry was the reason I spent many hours trying to get my solution to work.